common/artifact-manager/src/main/java/com/vmware/pscoe/iac/artifact/store/vrang/VraNgPropertyGroupStore.java

```diff
@@ -60,68 +72,107 @@ public void init(RestClientVraNg restClient, Package vraNgPackage, Configuration
-	 * so it gets its id and based on that performs either create or update operation.
-	 * If a property group doesn't exist set project ID in request.
-	 * If a property group exists, set the property group id in the request, so vRA knows which PG to update
+	 * Imports or update a specific property group.
 	 *
+	 * Note: a property group cannot be deleted immediately before import or update.
+	 * Because when property group delete is requested, the property group is only
+	 * marked for deletion.
+	 * Then it is deleted by the server at a later unspecified time.
+	 * While the property group exists and is marked for deletion, all update
+	 * operations succeed,
+	 * however, all get operations for this property group fail.
+	 * There is no way to restore such a property group, you have to wait for the
+	 * deletion, and then create it again.
+	 * 
+	 * 
+	 * Therefore, if deleting everything from an environment with the purpose of
+	 * having a clean slate for import,
+	 * one should make sure that enough time has passed, and everything is really
+	 * deleted.
+	 * 
 	 * @param propertyGroup
 	 * @param propertyGroupsFolder
+	 * @throws Exception
 	 */
-	private void importPropertyGroup(VraNgPropertyGroup propertyGroup, File propertyGroupsFolder) {
+	private void importPropertyGroup(VraNgPropertyGroup propertyGroup, File propertyGroupsFolder)
+			throws UnsupportedOperationException {
 		File customPropertyGroupFile = Paths.get(propertyGroupsFolder.getPath(), getName(propertyGroup)).toFile();
-		VraNgPropertyGroup existingPropertyGroup = this.getPropertyGroupByName(propertyGroup.getName());
+		String organizationId = VraNgOrganizationUtil.getOrganization(this.restClient, this.config).getId();
 
+		VraNgPropertyGroup existingPropertyGroup = this.getPropertyGroupByName(propertyGroup.getName());
+		// Replace existing organisation id with the id of target organisation
+		propertyGroup.setOrgId(organizationId);
 		if (existingPropertyGroup != null) {
-			if (!existingPropertyGroup.hasTheSameProjectId(this.projectId)) {
-				logger.warn("The property group {} is already assigned to a different project. Property group scope will NOT be changed and the group will NOT be updated.", getName(propertyGroup));
-			} else {
-				propertyGroup.setId(existingPropertyGroup.getId());
-				logger.info("Updating property group: {}", customPropertyGroupFile.getAbsolutePath());
-				this.restClient.updatePropertyGroup(propertyGroup);
+			// Property group with the same name already exists on the server
+			// Update
+			// First check for change of scope:
+			// If a non-blank projectID is different from the configuration project ID, we
+			// will request change of scope and the API will return an error.
+			// Scope change is not an operation supported by the API.
+			String existingPropertyGroupProjectId = existingPropertyGroup.getProjectId();
+			if (existingPropertyGroupProjectId != null && !existingPropertyGroupProjectId.isBlank()
```

Comment:
Let's use if, else if, else, top make things easier to understand. Consider also moving the "if" conditions to a variable like:

```java
Boolean belongsToOtherProject = existingPropertyGroupProjectId != null && !existingPropertyGroupProjectId.isBlank() && !existingPropertyGroupProjectId.equals(this.projectId)
```